src/user_api.ts

```diff
@@ -0,0 +1,534 @@
+  
+      if (lastEvaluatedKey) {
+        lastEvaluatedKeyQuery = `?lastEvaluatedKey=${lastEvaluatedKey}`;
+      }
+  
+      do {
+        if (timeoutMs !== undefined && Date.now() - startTime >= timeoutMs) {
+          this.logger.error('SyncMerkleRoots operation timed out');
+          throw new ErrorSyncMerkleRootsTimeout();
+        }
+        merkleRootsResponse = await this.http.request(`${requestPath}${lastEvaluatedKeyQuery}`, 'GET');
+  
+        if (previousLastEvaluatedKey === merkleRootsResponse.page.lastEvaluatedKey) {
+          this.logger.error(
+            'The last evaluated key has not changed between requests, indicating a possible loop or synchronization issue.',
+          );
+          throw new ErrorStaleLastEvaluatedKey();
+        }
+  
+        await repo.saveMerkleRoots(merkleRootsResponse.content);
+  
+        lastEvaluatedKeyQuery = `?lastEvaluatedKey=${merkleRootsResponse.page.lastEvaluatedKey}`;
+        previousLastEvaluatedKey = merkleRootsResponse.page.lastEvaluatedKey;
+      } while (previousLastEvaluatedKey !== '');
+    }
+  
+    /**
+     * Generate TOTP for a contact
+     *
+     * @param {Contact} contact - Contact to generate TOTP for
+     * @param {number} [period=DEFAULT_TOTP_PERIOD] - TOTP period
+     * @param {number} [digits=DEFAULT_TOTP_DIGITS] - Number of TOTP digits
+     * @returns {string} Generated TOTP
+     * @throws {ErrorNoXPrivToGenerateTOTP} If xPriv is not available
+     */
+    generateTotpForContact(
+      contact: Contact,
+      period: number = DEFAULT_TOTP_PERIOD,
+      digits: number = DEFAULT_TOTP_DIGITS,
+    ): string {
+      if (!this.xPriv) {
+        throw new ErrorNoXPrivToGenerateTOTP();
+      }
+      return generateTotpForContact(this.xPriv, contact, period, digits);
+    }
+  
+    validateTotpForContact(
+      contact: Contact,
+      passcode: string,
+      requesterPaymail: string,
+      period: number = DEFAULT_TOTP_PERIOD,
+      digits: number = DEFAULT_TOTP_DIGITS,
+    ): boolean {
+      if (!this.xPriv) {
+        throw new ErrorNoXPrivToValidateTOTP();
+      }
+      return validateTotpForContact(this.xPriv, contact, passcode, requesterPaymail, period, digits);
+    }
+  
+    /**
+     * Get a list of paymail addresses
+     *
+     * @param {PaymailFilter} conditions - Filter conditions for paymail addresses
+     * @param {Metadata} metadata - Metadata filter
+     * @param {QueryPageParams} queryParams - Pagination parameters
+     * @returns {Promise<PageModel<PaymailAddress>>} List of paymail addresses
+     */
+    async paymails(
+      conditions: PaymailFilter,
+      metadata: Metadata,
+      queryParams: QueryPageParams,
+    ): Promise<PageModel<PaymailAddress>> {
+      const basePath = 'paymails';
+      const queryString = buildQueryPath({
+        filter: conditions,
+        metadata,
+        page: queryParams,
+      });
+  
+      return await this.http.request(`${basePath}${queryString}`, 'GET');
+    }
+  }
+  
```

Comment:
Remember to add empty line at the end of the files.
[Here](https://stackoverflow.com/a/44704969) is some configuration instruction for Visual Studio Code to do that automatically